packages/ra-core/src/dataProvider/useUpdate.ts

```diff
@@ -45,74 +55,319 @@ import useMutation, { MutationOptions, Mutation } from './useMutation';
+
+                if (reactMutationOptions.onError) {
+                    return reactMutationOptions.onError(
+                        error,
+                        variables,
+                        context
+                    );
+                }
+            },
+            onSuccess: (
+                data: RecordType,
+                variables: Partial<UseUpdateParams<RecordType>> = {},
+                context: unknown
+            ) => {
+                if (mutationMode === 'pessimistic') {
+                    // Optimistically update to the new value in getOne
+                    const {
+                        resource: callTimeResource = resource,
+                        id: callTimeId = id,
+                    } = variables;
+                    updateCache({
+                        resource: callTimeResource,
+                        id: callTimeId,
+                        data,
+                    });
+
+                    if (reactMutationOptions.onSuccess) {
+                        return reactMutationOptions.onSuccess(
+                            data,
+                            variables,
+                            context
+                        );
+                    }
+                }
+            },
+            onSettled: (
+                data: RecordType,
+                error: unknown,
+                variables: Partial<UseUpdateParams<RecordType>> = {},
+                context: unknown
+            ) => {
+                const {
+                    resource: callTimeResource = resource,
+                    id: callTimeId = id,
+                } = variables;
+                if (
+                    mutationMode === 'optimistic' ||
+                    mutationMode === 'undoable'
+                ) {
+                    // Always refetch after error or success:
+                    queryClient.invalidateQueries([
+                        callTimeResource,
+                        'getOne',
+                        callTimeId,
+                    ]);
+                    queryClient.invalidateQueries([
+                        callTimeResource,
+                        'getList',
+                    ]);
+                }
+
+                if (reactMutationOptions.onSettled) {
+                    return reactMutationOptions.onSettled(
+                        data,
+                        error,
+                        variables,
+                        context
+                    );
+                }
+            },
+        }
     );
 
-    return [update, state];
+    const mutateAsync = async (
+        variables: Partial<UseUpdateParams<RecordType>> = {},
+        callTimeOptions: MutateOptions<
+            RecordType,
+            unknown,
+            Partial<UseUpdateParams<RecordType>>,
+            unknown
+        > = {}
+    ) => {
+        if (mutationMode === 'pessimistic') {
+            return mutation.mutateAsync(variables, callTimeOptions);
```

Comment:
If I understand correctly, we don't merge the declaration time options with the calltime ones?

Reply:
this is done by the mutation callback (the first argument of useMutation)